LabExT/View/Movement/LoadStoredCalibrationWindow.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+LabExT  Copyright (C) 2022  ETH Zurich and Polariton Technologies AG
+This program is free software and comes with ABSOLUTELY NO WARRANTY; for details see LICENSE file.
+"""
+import json
+
+from tkinter import W, Frame, Label, OptionMenu, StringVar, Toplevel, Button, messagebox, RIGHT, TOP, X, BOTH, FLAT, Y, LEFT
+from typing import Type
+from os.path import exists
+
+from LabExT.Utils import run_with_wait_window
+from LabExT.Movement.MoverNew import MoverNew
+from LabExT.View.Controls.CustomFrame import CustomFrame
+from LabExT.View.Controls.DeviceTable import DeviceTable
+from LabExT.Wafer.Chip import Chip, Device
+
+
+class LoadStoredCalibrationWindow(Toplevel):
+    """
+    Window to load stored calibrations
+    """
+
+    ASSIGNMENT_MENU_PLACEHOLDER = "-- unused --"
+
+    def __init__(
+        self,
+        master,
+        mover: Type[MoverNew],
+        chip: Type[Chip]
+    ) -> None:
+        """
+        Parameters
+        ----------
+        master : Tk
+            Tk instance of the master toplevel
+        mover : Mover
+            Instance of the current mover.
+        mover : Chip
+            Instance of the current chip.
+
+        Raises
+        ------
+        RuntimeError
+            If calibrations could not be restored.
+        """
+        self.mover: Type[MoverNew] = mover
+        self.chip: Type[Chip] = chip
+
+        if not messagebox.askyesno(
+            "Restore calibration",
+                f"Found mover calibration for chip: {self.chip.name}. Do you want to restore it?"):
+            return
+
+        self.calibration_settings = None
+        if not exists(self.mover.calibration_settings_file):
+            raise RuntimeError(
+                "Calibration settings file does not exits.")
+
+        with open(self.mover.calibration_settings_file, "r") as fp:
+            self.calibration_settings = json.load(fp)
+
+        self.stored_calibrations = self.calibration_settings.get(
+            "calibrations", [])
+        if len(self.stored_calibrations) == 0:
+            raise RuntimeError(
+                f"Calibration settings file {self.calibration_settings} does not contain any calibrations.")
+
+        self.calibration_options = [
+            f"Orientation: {c.get('orientation')} - Port: {c.get('device_port')}" for c in self.stored_calibrations]
+        self.calibration_options.append(self.ASSIGNMENT_MENU_PLACEHOLDER)
+
+        if self.calibration_settings.get("chip_name") != self.chip.name:
+            if not messagebox.askokcancel(
+                "Confirm restoring.",
+                f"The calibration file {self.mover.calibration_settings_file} was calibrated "
+                f"with chip {self.calibration_settings.get('chip_name')}. "
+                    f"You imported chip {self.chip.name}. Are you sure to continue?"):
```

Comment:
Suggestion: include the timestamp on when the calibration was created and saved. This helps the user to see if he needs to re-do it, e.g. when coming in again in the morning.

Reply:
Added :)